frontend/src/routes/__root.tsx

```diff
@@ -9,5 +13,24 @@ export const Route = createRootRoute({
       <TailwindIndicator />
       <TanStackRouterDevtools />
     </main>
-  )
+  ),
+  notFoundComponent: withAuthRedirect(NotFound)
```

Comment:
```suggestion
  notFoundComponent: NotFound
```

We don't want to redirect from 404 if the user is not connected, we don't care.